### ISSUE TITLE
Some restrictions fail to set if other restrictions are already set.

### ISSUE THREAD

**Issue by Medea-Destiny** [OpenCollarTeam/OpenCollar#664]
In 8.1, go to Restrictions, Talk, and activate EmoteTrunc. Now come out of talk and go to movement, and activate Fly. 

Nothing happens.

Okay, bear with me a minute, this gets technical. When that fly button is clicked, the oc_rlvsuite script parses out the function name, looks for the index of that string in the list restrictions, and figures out the two-byte bitmask value of that restriction. Then it checks to see if it is currently set and if so unsets it, or otherwise sets it.

The problem is that it needs to know which byte to compare the bitmask to. The current code attempts to do this by zeroing the first mask byte if the index is in the second mask. Then it does an AND comparison of the mask for the two bytes. The problem is that if the byte 1 bitmask is non-zero, it will return TRUE for an AND with zero.

In short, if any of the first 32 restrictions have been set, then attempting to set any of the second 32 restrictions will fail, because the code thinks it's already set, so it sends an unset command rather than a set command.

The solution to this is to change the way this code operates when a dialog response is provided. As the function that provides a category menu already checks to see if a restriction is set to provide the checkbox, we can just read the checkbox to determine whether the function is set and send the button name sans checkbox with the checkbox bit value directly to the applycommand function.

This fix is included in my work for meta-issue #649
